File: component-acgnews/src/main/java/com/rabtman/acgnews/mvp/contract/ZeroFiveNewsContract.java
package com.rabtman.acgnews.mvp.contract;

import com.rabtman.acgnews.mvp.model.jsoup.ZeroFiveNews;
import com.rabtman.acgnews.mvp.model.jsoup.ZeroFiveNewsPage;
import com.rabtman.common.base.mvp.IModel;
import com.rabtman.common.base.mvp.IView;
import io.reactivex.Flowable;
import java.util.List;

/**
 * @author Rabtman
 */
public interface ZeroFiveNewsContract {

  interface View extends IView {

    String getNewsUrl(int pageNo);

    void showAcgNews(List<ZeroFiveNews> zeroFiveNewsList);

    void showMoreAcgNews(List<ZeroFiveNews> zeroFiveNewsList, boolean canLoadMore);

    void onLoadMoreFail();
  }

  interface Model extends IModel {

    Flowable<ZeroFiveNewsPage> getAcgNews(String typeUrl);
  }
}

File: component-acgnews/src/main/java/com/rabtman/acgnews/mvp/contract/ZeroFiveNewsContract.kt
package com.rabtman.acgnews.mvp.contract

import com.rabtman.acgnews.mvp.model.jsoup.ZeroFiveNews
import com.rabtman.acgnews.mvp.model.jsoup.ZeroFiveNewsPage
import com.rabtman.common.base.mvp.IModel
import com.rabtman.common.base.mvp.IView
import io.reactivex.Flowable

/**
 * @author Rabtman
 */
interface ZeroFiveNewsContract {
    interface View : IView {
        fun getNewsUrl(pageNo: Int): String
        fun showAcgNews(zeroFiveNewsList: List<ZeroFiveNews>)
        fun showMoreAcgNews(zeroFiveNewsList: List<ZeroFiveNews>?, canLoadMore: Boolean)
        fun onLoadMoreFail()
    }

    interface Model : IModel {
        fun getAcgNews(typeUrl: String?): Flowable<ZeroFiveNewsPage>
    }
}